core/jazz_environment-event-handler/index.js

```diff
@@ -591,6 +624,214 @@ function triggerBuildJob(payload, serviceDetails, configData) {
+      method: "POST",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+        "Jazz-Service-ID": service_id
+      },
+      json: updatePayload,
+      rejectUnauthorized: false
+    };
+    logger.info("safe svcpayload ****" + JSON.stringify(svcPayload));
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        logger.info("response ***" + JSON.stringify(response));
+        logger.info("enviornmnet***" + JSON.stringify(environmentPayload));
+        const timestamp = new Date().toISOString();
+        const vaultLink = configData.TVAULT.HOSTNAME + "/#!/admin";
+        environmentPayload.safeDetails = { "safe": { "name": safeName, "ts": timestamp, "link": vaultLink } };
+        safeDetails = environmentPayload.safeDetails;
+        logger.info("environmentPayload createsafe" + JSON.stringify(environmentPayload));
+        return resolve(environmentPayload);
+      } else {
+        logger.error("Error creating safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error creating safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function deleteSafe(environmentPayload, service_id, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT && !configData.TVAULT.IS_ENABLED) {
+        logger.info('t-vault not enabled **', configData.TVAULT);
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    let safeName;
+    if (safeDetails && safeDetails.safe && safeDetails.safe.name) {
+      safeName = safeDetails.safe.name;
+    }
+    var svcPayload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + configData.TVAULT.CREATE_SAFE_API_RESOURCE + '/' +safeName,
+      method: "DELETE",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+      },
+      rejectUnauthorized: false
+    };
+    logger.info("safe svcpayload ****" + JSON.stringify(svcPayload));
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        logger.info("response ***" + JSON.stringify(response));
+        return resolve(body);
+      } else {
+        logger.error("Error deleting safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error deleting safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function getAdmins(environmentPayload, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT && !configData.TVAULT.IS_ENABLED) {
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+
+    var payload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + '/jazz/usermanagement/users',
```

Comment:
Please get the api information from the config files

Reply:
Added APIs endpoint in config.json file.